index.js

```diff
@@ -7,12 +7,55 @@
 
 'use strict';
 
-module.exports = function dashify(str) {
-  if (typeof str !== 'string') {
+module.exports = dashify
+
+function dashify(string) {
+  if (typeof string !== 'string') {
     throw new TypeError('expected a string');
   }
-  str = str.replace(/([a-z])([A-Z])/g, '$1-$2');
-  str = str.replace(/[ \t\W]/g, '-');
-  str = str.replace(/^-+|-+$/g, '');
-  return str.toLowerCase();
+
+  var sentence = []
+  var needsDash = false
+
+  for(var i=0; i<string.length; i++) {
```

Comment:
Hi @erikpukinskis, thanks for the pr! would you mind fixing code formatting to be consistent with the rest of the code? (whitespace and semicolons mostly. if you're unsure you can run eslint. thanks in advance)